core/modules/widgets/navigator.js

```diff
@@ -539,8 +540,8 @@ NavigatorWidget.prototype.handleImportTiddlersEvent = function(event) {
 	// Save the $:/Import tiddler
 	newFields.text = JSON.stringify(importData,null,$tw.config.preferences.jsonSpaces);
 	this.wiki.addTiddler(new $tw.Tiddler(importTiddler,newFields));
-	// Update the story and history details
-	var autoOpenOnImport = event.autoOpenOnImport ? event.autoOpenOnImport : this.getVariable("tv-auto-open-on-import");  
+	// Update the story and history details. Note: paramObject is either event or event.paramObject
```

Comment:
I don't think the logic is quite right here. The value of paramObject only falls back to the event object if the paramObject is entirely missing. We need to do that check at the next level down: something like `event.importTitle || event.paramObject.importTitle || IMPORT_TITLE".

Reply:
Thanks, @Jermolene. This would be to guard against things going wrong when dispatchEvent is called directly, right?

Here is how dropzone.js is calling it:

```js
this.dispatchEvent({
  type: "tm-import-tiddlers",
  param: JSON.stringify(tiddlerFieldsArray),
  autoOpenOnImport: this.autoOpenOnImport,
  importTitle: this.importTitle
});
```

I think if instead someone did the following (last line is different), then your suggested change would be required for it to work.

```js
this.dispatchEvent({
  type: "tm-import-tiddlers",
  param: JSON.stringify(tiddlerFieldsArray),
  autoOpenOnImport: this.autoOpenOnImport,
  paramObject: {importTitle: this.importTitle}
});
```

Reply:
Thanks @btheado, that's correct.

We may well want in the future to introduce the ability to dispatch events with custom properties on the root of the event object because it would be required for full backwards compatibility as we move more functionality from JavaScript into wikitext (eg the navigator widget).

Reply:
Thanks, @Jermolene. I changed the code. Now the parameters can be split between the top level event object and paramObject and it should still work.